src/com/cloudogu/ces/cesbuildlib/Git.groovy

```diff
@@ -248,13 +266,28 @@ class Git implements Serializable {
      * This method executes the git command with a bash function as credential helper,
      * which return username and password from jenkins credentials.
      *
+     * If the script failes with exit code 128, this will retry the call up to 5 times
+     * before failing.
+     *
      * @param args git arguments
      */
     protected void executeGitWithCredentials(String args) {
         if (credentials) {
             script.withCredentials([script.usernamePassword(credentialsId: credentials,
                     passwordVariable: 'GIT_AUTH_PSW', usernameVariable: 'GIT_AUTH_USR')]) {
-                script.sh "git -c credential.helper=\"!f() { echo username='\$GIT_AUTH_USR'; echo password='\$GIT_AUTH_PSW'; }; f\" ${args}"
+                def pushResultCode = 128
+                def retryCount = 0
+                while (pushResultCode == 128 && retryCount < maxRetries) {
+                    if (retryCount > 0) {
+                        script.echo "Got error code ${pushResultCode} - retrying in ${retryTimeout} ms ..."
+                        sleep(retryTimeout)
```

Comment:
I'm not sure if the pipeline [sleep](https://jenkins.io/doc/pipeline/steps/workflow-basic-steps/#sleep-sleep) step is better suited for this

Reply:
Is this a real method? or do you mean `script.sleep`?

Reply:
This is the groovy sleep. Wasn't aware of a pipeline version